IPython/core/tests/test_completer.py

```diff
@@ -294,14 +294,26 @@ def _test_not_complete(reason, s, comp):
 
     import jedi
     jedi_version = tuple(int(i) for i in jedi.__version__.split('.')[:3])
-    if jedi_version > (0,10):
+    if jedi_version > (0, 10):
         yield _test_complete, 'jedi >0.9 should complete and not crash', 'a=1;a.', 'real'
     yield _test_complete, 'can infer first argument', 'a=(1,"foo");a[0].', 'real'
     yield _test_complete, 'can infer second argument', 'a=(1,"foo");a[1].', 'capitalize'
     yield _test_complete, 'cover duplicate completions', 'im', 'import', 0, 2
 
     yield _test_not_complete, 'does not mix types', 'a=(1,"foo");a[0].', 'capitalize'
 
+def test_deduplicate_completions():
+    """
+    Test that completions are correctly deduplicated (even if ranges are not the same)
+    """
+    ip = get_ipython()
+    ip.ex('import itertools as it')
```

Comment:
I would instead write the test to run with a simpler object, like a simple custom class:

```
In [5]: class Z:
   ...:     z = 1
   ...:     

In [6]: Z.z<TAB>
```

If itertools later develops a second method starting with the letter 'a', this test would break.